### ISSUE TITLE
Updating three.JS loading manager on progress

### ISSUE THREAD

**Issue by DigDigDig** [minorua/Qgis2threejs#91]
The generated .js file of the DEM and layer content is quite large, over even fast connections it takes a while for the browser to load the full data package and render the scene.  In the past, I've utilized the [three.js loading manager](https://threejs.org/docs/#Reference/Loaders.Managers/DefaultLoadingManager) to update a progress bar when loading my objects and texture in scenes.  Because this plugin uses a lot of custom loaders, I'm finding it difficult to navigate where to pass (manager) through.  

Is the three.js manager utility compatible with the method of loading that Qgis2threejs uses?

**Comment by minorua** [comment]
DEM data, texture images and vector data are all output to a .js file (or .js files). This is a workaround for Chrome's restriction on accessing local files.

In future, the plugin will have a web view compatible with WebGL and be possible to display 3D scene inside it. It will have a menu command "Publish to Web". The output data files are individual (binary, JSON, etc.) so that asynchronous loading is possible. I think probably the loading manager is useful.

**Comment by DigDigDig** [comment]
Thank you minorua.  I've been having a lot of fun digging around in the generated files.  Very happy that you found a way to circumvent the cross-origin request error.  being able to run the HTML file locally without going through a virtual server is a valuable feature.  I found a workaround for updating a progress bar which does help a lot in cases of long load times.

Always looking forward to seeing what changes are to come.

Thanks for the hard work!

**Comment by nmalawskey** [comment]
DigDigDig running into a similar issue, would you be willing to share your workaround?

Thanks!

**Comment by DigDigDig** [comment]
Sure thing.  I took a two prong approach using [Pace.js](http://github.hubspot.com/pace/docs/welcome/) and a custom spinner.

![image](https://user-images.githubusercontent.com/17930616/36693277-ce1be47e-1b08-11e8-97fa-173e71d97808.png)

Pace is that top blue bar, it's SUPER easy to install.

The spinner is just an html element with some css animation and has a black div behind it.  I wrote a short function in the qgis2threejs.js file that would hide the spinner when I wanted it to go away after the data had loaded.


```

var loadingData = document.getElementById('loading-data');

function hideLoader() {
    loadingData.style.visibility = 'hidden';
}
```

Then I just called the `hideLoader()` function from app.start:

```
 app.start = function () {
        app.running = true;
        if (app.controls) app.controls.enabled = true;
        app.animate();
        hideSpinner();
    };
```
